prebuild/emscripten/blst.d.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  P1_AffineConstructor,
+  P1Constructor,
+  P2_AffineConstructor,
+  P2Constructor,
+  PairingConstructor,
+  PTConstructor,
+  SecretKeyConstructor,
+} from "../../src/bindings";
+
+export const initialized: Promise<void>;
+
+// BLS12_381_G1: P1_Affine;
+// BLS12_381_NEG_G1: P1_Affine;
+// BLS12_381_G2: P2_Affine;
+// BLS12_381_NEG_G2: P2_Affine;
```

Comment:
I'd suggest not aiming for providing access to these. There are `generator()` methods that serve the purpose. Though one can make a case for constant generators. I mean `generator()` methods create mutable copies... Which can be viewed as an extra burden, because as far as I understand you're responsible for freeing all the objects explicitly. One can reformulate the question as following. Are there cases when constant generator points would be ultimately useful? One common case is the `PT` constructor, but it should be noted that the underlying subroutine, `blst_miller_loop`, actually replaces null references with generators...

Reply:
This one has me in a bit of an awkward position. :sweat_smile: My original intention was to replace the native binding, leaving the API layer unchanged. I don't know enough to have strong opinion on this but perhaps @dapplion can weigh in here?

Reply:
Well, sometimes one has to recognize the limitations. I mean webasm interface is purely functional and one can't share constant data in the same way. As already mentioned, there are `generator()` methods and they are available even in native bindings. So that it's possible to adapt the existing code to avoid the limitation of one... But sure, let's first hear how bad is it. In worst case one can define them as constants, and have C++ side recognize them and replace with pointers to constant data...

Reply:
Is there are way to execute some code upon module load? There should be... It looks like it would be possible to instantiate the objects by having contructors recognize the constants and returning pointers to constant data... Though problem would be that the data won't be actually constant. I mean unlike the native bindings where attempt to manipulate would trigger a crash, here a buggy program would get away with otherwise erroneous modification, which would incur hard-to-find bugs later on... Again, `generator()` methods would be preferable... But again, let's first hear...

Reply:
> Though problem would be that the data won't be actually constant... hard-to-find bugs later on...

On closer examination maybe it's not that bad. In sense that the constant points in question are affine and the affine class doesn't have methods that would modify the data. In other words an application won't have an easily accessible way to wreck havoc...

Reply:
(additional lifecycle hooks are also available)

Reply:
> This one has me in a bit of an awkward position. sweat_smile My original intention was to replace the native binding, leaving the API layer unchanged. I don't know enough to have strong opinion on this but perhaps @dapplion can weigh in here?

Since this library is eventually wrapped by chainsafe/bls I don't think it necessary that bindings and wasm API as the same. If wasm exports only a functional API that fits consumer ergonomics, then that's fine.

Reply:
> (additional lifecycle hooks are also available)

Could you provide some pointers? Like a real-life usage example and a specification or something...

Reply:
Unfortunately, no. Usage of the `Module.onRuntineInitialized` hook and the emscripten docs that I referenced above contain the full extent of my understanding at the moment.